react-components/src/components/SceneContainer/SceneContainer.tsx

```diff
@@ -0,0 +1,240 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { type ReactElement, useEffect, useRef, useState } from 'react';
+import { type AddModelOptions } from '@cognite/reveal';
+import { useReveal } from '../RevealContainer/RevealContext';
+import * as THREE from 'three';
+import { type Query } from '../../utilities/FdmSDK';
+import { useFdmSdk } from '../RevealContainer/SDKProvider';
+import { createGetSceneQuery } from './Queries';
+import {
+  type GroundPlaneResponse,
+  type GroundPlaneProperties,
+  type SceneResponse,
+  type SkyboxProperties,
+  type GroundPlaneEdgeResponse,
+  type Transformation3d,
+  type SceneModelsProperties
+} from './SceneFdmTypes';
+import { type AddResourceOptions, Reveal3DResources } from '../..';
+import type CogniteClient from '@cognite/sdk/dist/src/cogniteClient';
+
+export type CogniteSceneProps = {
+  sceneExternalId: string;
+  sceneSpaceId: string;
+  sdk: CogniteClient;
+  onResourcesAdded?: () => void;
+  onResourceLoadError?: (error: any) => void;
+};
+
+export function SceneContainer({
+  sceneExternalId,
+  sceneSpaceId,
+  sdk
```

Comment:
The API for this component should be an extension of `Reveal3DResources` component, i.e. support styling of instances and callback for when models are loaded.

Reply:
The reason for that is FDX and other apps relies on styling capabilities a lot.